app/src/main/java/com/github/campus_capture/bootcamp/fragments/MapsFragment.java

```diff
@@ -153,11 +188,93 @@ public View onCreateView(@NonNull LayoutInflater inflater,
     public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
         super.onViewCreated(view, savedInstanceState);
 
+        zoneDB = Room.databaseBuilder(getActivity(),
+                        ZoneDatabase.class, "zones-db")
+                .createFromAsset("databases/zones-db.db")
+                .allowMainThreadQueries().build();
+
         SupportMapFragment mapFragment =
                 (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
         if (mapFragment != null) {
             mapFragment.getMapAsync(callback);
         }
+
+        scheduler = new MapScheduler(view, backendInterface, this);
+
+        view.findViewById(R.id.attackButton).setOnClickListener(attackListener);
+        view.findViewById(R.id.defendButton).setOnClickListener(attackListener);
+
+        scheduler.startAll();
+    }
+
+    @Override
+    public void onDestroyView() {
+        scheduler.stopAll();
+        zoneDB.close();
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onDestroy()
+    {
+        zoneDB.close();
+        super.onDestroy();
+    }
+
+    /**
+     * Method which finds in which zone a given point is
+     * @param position the position
+     * @return the (first) zone which matches the position in the DB, or null if none
+     */
+    public Zone findCurrentZone(LatLng position)
+    {
+        ZoneDAO zoneDAO = zoneDB.zoneDAO();
+        List<Zone> zones = zoneDAO.getAll();
+        for(Zone z : zones)
+        {
+            if(PolyUtil.containsLocation(position, z.getVertices(), false))
+            {
+                return z;
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Method to get the current position of the player. Note that this uses a deprecated method
+     * of map, and should use a FusedLocationProviderClient instead
+     * @return LatLng or null
+     */
+    public LatLng getCurrentPosition()
+    {
+        if(!locationOverride)
+        {
+            Location loc = null;
+            try
+            {
+                if (!permissionDenied)
+                {
+                    loc = map.getMyLocation();
+                }
+            }
+            catch (Exception e)
+            {
+                Log.e("MapsFragment", "Failed retrieving location");
+            }
+
+            if(loc == null)
+            {
+                return null;
+            }
+            else
+            {
+                return new LatLng(loc.getLatitude(), loc.getLongitude());
+            }
```

Comment:
could use ternary operator to satisfy codeclimate btw

Reply:
Well spotted